File: python/desafios/tomadas.py
print("---------------------DESAFIO: TOMADAS----------------------\n")

values_isOk = False

while not values_isOk:
    
    print("--------------------------------------------------")

    rules_list_not_filtered = list(input("Digite o número de tomadas de cada uma das 4 reguas: "))
    rules_list_filtered = list()

    for i in range(len(rules_list_not_filtered)):
        try:
            socket_numbers = int(rules_list_not_filtered[i])
            
            #Verifica se o número de tomadas de cada valor está correto
            if socket_numbers < 2 or socket_numbers > 6:
                print("O número de tomadas não pode ser menor que 2 e nem exceder 6!!!")
                break
            
            # Caso não seja a ultima regua, faz a correção da extensão
            if i != len(rules_list_not_filtered)-1:
                socket_numbers-=1
                
            #Caso tudo esteja certo adiciona o valor a nova lista
            rules_list_filtered.append(socket_numbers)
                
        #tratamentos de espaços e caracteres que não podem ser convertidos em números
        except TypeError:
            pass
        except ValueError:
            pass
    
    #Validação do números de elementos na listai
    if len(rules_list_filtered) != 4:
        print("Devem ser digitados 4 números de tomadas!")
    else: #Caso passe pela ultima validação indica como True, pois todos os valores estão agora correntos
        values_isOk = True

#finalização do programa
print("----------------------------------------------------------\n")
print(sum(rules_list_filtered))

File: java/desafios/tomadas.java
package desafios;

import java.util.Scanner;
import libraries.TerminalFormating;

public class tomadas {

    public static void main(String[] args) {

        boolean values_isOk = false;
        int rules_list[] = new int[4];
        int socket = 0;
        int sum = 0;

        Scanner input = new Scanner(System.in);
        TerminalFormating terminal = new TerminalFormating();

        terminal.writeProgramTitle("Tomadas");

        while (!values_isOk) {
            System.out.println("Digite a quantidade de tomadas de cada uma das 4 reguas separado por espaço: ");
            String[] info = input.nextLine().split(" ");
            
            if (info.length != 4) {
                terminal.errorMessage("Deve ter 4 valores!");
                break;
            }

            try {
                for (int i = 0; i < 4; i++){
                    socket = Integer.parseInt(info[i]);
    
                    if (socket < 2 || socket > 6) {
                        terminal.errorMessage("O número de tomadas não pode ser menor 2 ou exceder 6!!!");
                        break;
                    }

                    if (i != 3) {
                        rules_list[i] = socket-1;
                    } else {
                        rules_list[i] = socket;
                    }
                }
        
            } catch(NumberFormatException ex) {
                terminal.errorMessage("O número não pode ser quebrado!");
                continue;
            }

            values_isOk = true;
        }

        for(int rule : rules_list) {
            sum+=rule;
        }

        System.out.println("----------------------------------");
        System.out.println(sum);
    }
}

  
